cfme/tests/infrastructure/test_host_analysis.py

```diff
@@ -19,110 +14,78 @@
 
 
 def pytest_generate_tests(metafunc):
-    argnames, argvalues, idlist = testgen.providers_by_class(metafunc, [InfraProvider])
+    argnames, argvalues, idlist = testgen.providers_by_class(metafunc, [InfraProvider],
+                                                             required_fields=['hosts'])
     argnames = argnames + ['host_type', 'host_name']
     new_argvalues = []
     new_idlist = []
 
-    for i, argvalue_tuple in enumerate(argvalues):
+    for index, argvalue_tuple in enumerate(argvalues):
         args = dict(zip(argnames, argvalue_tuple))
 
-        prov_hosts = args['provider'].data.get('hosts')
+        prov_hosts = args['provider'].data.hosts
 
-        if not prov_hosts:
-            continue
         for test_host in prov_hosts:
             if not test_host.get('test_fleece', False):
                 continue
-            assert test_host.get('type') in HOST_TYPES,\
-                'host type must be set to [{}] for smartstate analysis tests'\
-                .format('|'.join(HOST_TYPES))
+            assert test_host.get('type') in HOST_TYPES, (
+                'host type must be set to [{}] for smartstate analysis tests'.format(
+                    '|'.join(HOST_TYPES)))
 
-            argvalues[i] = argvalues[i] + [test_host['type'], test_host['name']]
-            test_id = '{}-{}-{}'.format(args['provider'].key, test_host['type'], test_host['name'])
-            new_argvalues.append(argvalues[i])
+            argvalues[index] = argvalues[index] + [test_host['type'], test_host['name']]
+            test_id = '{}-{}'.format(args['provider'].key, test_host['type'])
+            new_argvalues.append(argvalues[index])
             new_idlist.append(test_id)
     testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope="module")
 
 
-def get_host_data_by_name(provider_key, host_name):
-    for host_obj in conf.cfme_data.get('management_systems', {})[provider_key].get('hosts', []):
-        if host_name == host_obj['name']:
-            return host_obj
-    return None
-
+@pytest.fixture(scope='module')
+def host_with_credentials(appliance, provider, host_name):
```

Comment:
**REQUIRED**
You can use this function https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/hosts.py#L21

Reply:
Yes, it is an option, but I would rather update cheds via API here

Reply:
I'm okay with this since the utils method doesn't use rest.